cmake/sdp.cmake

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+function(sdp_assembly_generate hrt_srcs)
+  set(hrt_msg "Generating ASM files for Hard Real Time files.")
+  set(hrt_opts -g0 -P -fno-ident -fno-verbose-asm)
```

Comment:
why `-P` ?
You are stopping after compiler proper, not after the preprocessor, so there should be no line markers in the assemply.
Ref: https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Preprocessor-Options.html
> `-P` Inhibit generation of linemarkers in the output from the preprocessor. This might be useful when running the preprocessor on something that is not C code, and will be sent to a program which might be confused by the linemarkers.

Reply:
I removed the `-P` because the code looks the same with and without it, and the comments are still there as shown in the comment below.

Reply:
> To generate asm output we use the -S option, not -E.

yes, I know, I was just curious if you during development had been using `-E`, in which case you normally see the `# <lineno> "<path>"`
It's common to see people experiencing issues during development with certain flags, and thus introducing fixes.
Fixes which later becomes obsolete when later changes are introduced.

So was trying to avoid such case.

Are you using the `#APP` and `#NO_APP` identifiers in the original source file ?
Will ping ofline for more into.

Reply:
seems to be caused by inline assembly.
Please note that not every version of gcc will use `#`, you can also risk a `@`, for example like this:
```assembly
@ 14 "/projects/github/review/nrf/applications/sdp/gpio/src/hrt/hrt.c" 1                            
        .align 1                                                                                    
@ 0 "" 2                
```

Reply:
current investigation has found no options to prevent gcc from placing a file comment in generated assembly.

One risk with the post comment removal is the fact that gcc can use `#` or `@` as comment identifier.
One factor impacting gcc's choice seems to be the `-std=<std>` in use.

An alternative is to invoke the compiler from the source folder or west workspace, and then use the relatice path to the source file.

This can be done by setting `WORKING_DIRECTORY <dir>` and adjust the input source path accordingly.
This can result in input like:
```assembly
@ 14 "hrt.c" 1
	.align 1
@ 0 "" 2
```
or
```assembly
@ 14 "nrf/applications/sdp/gpio/src/hrt/hrt.c" 1
	.align 1
@ 0 "" 2
```

but for this PR, such change is not requested.
It's better to do such alternative in future followup.